### ISSUE TITLE
sandbox with github repo as parent does not show on profile > sandboxes/showcase

### ISSUE THREAD

**Issue by faceyspacey** [CompuIves/codesandbox-client#59]
For example, go here:

https://codesandbox.io/u/faceyspacey/likes

You will see my main codesandbox there, but it does not appear here:

https://codesandbox.io/u/faceyspacey/sandboxes

and also it cannot be selected as for "showcase."

**Comment by CompuIves** [comment]
You're right, there is no current way to map a github sandbox to a user.

It's definitely on the roadmap to add this functionality, though it's not here right now for the reason that the user shouldn't be able to edit the sandbox. I need to change functionality on the server and client to allow changes to the title/description/tags etc, but to disable editing files.

**Comment by faceyspacey** [comment]
Thought so. Would be nice to be getting credit for my codesandbox stats. Very important for developers serious about promoting their sandboxes.

...for the showcase maybe u could paste the URL of any sandbox. Or select from liked sandboxes as well.

**Comment by faceyspacey** [comment]
I need this brotha. as well as to show my logo in the sandbox instead of just the github logo. And the ability to have this appear on my profile etc.

I get that less users probably use this feature. But the most important sandboxes will. The github connection needs to be flawless for CodeSandBox to achieve the "seriousness" it is capable of.

If the github connection is propped up, made feature complete, etc, it will get more usage and become a key part of CodeSandBox. 

Basically syncing back to your repo from codesandbox will be the feature that helps this achieve the inflection point where it's a crucial aspect of the platform. What you want is to promote "EXPORT TO GITHUb" (or "FORK TO GITHUB") instead of import. This will make CodeSandBox go more viral than the embed buttons, as this is a frictionless activity users are already familiar how to do and doing it many times throughout the day.

It also wouldn't be a true fork. it's a creating a repo--so you could name the repo something with your URL it if want, and/or add anything you want to say to the readme, SUCH AS ADD YOUR MARKDOWN BUTTON! Or even better take a screenshot of the app, and put it at the top of the readme right below that button!!

There are tools to convert html to images via javascript in the browser. This is the primary tool if you don't already know it: https://html2canvas.hertzen.com/

Convert HTML to canvas, and then canvas to image. That's how it's done.

It can't capture iframes or certain kinds of photos with hot-loading blocked, etc. But that won't be the case most of the time inside your iframe. So you need to generate inside the app iframe, and then pass it back to the parent. You won't be able to generate it in the parent (i'm pretty sure).

So boom, now you're posting to repos an image, your branded button, whatever else you wanna say in the readme you create (or edit, if one already exists in the sandbox). It's one-click share instead of the embed modal, which is like 8 clicks:

- open modal
- choose 3 options
- go back go medium article
- click edit on medium article
- find place to put it.
- paste in link
- enter new line so embedly converts it
- publish article
- etc

That is a lot more friction than "FORK ON GITHUB". The latter will explode your popularity. 

So that's why this is an important feature brotha, and more than just giving me the things regular sandbox users have. Please don't wait until you do all that before giving repo-based sandboxes parity to regular sandboxes. Achieving parity will take far less time than all that other stuff. But you need to do it anyway, and you don't want users scared of creating repo-based sandboxes cuz it doesnt work as well. You probably want to find a root-level way to make it so repo-based sandboxes aren't that different from regular sandboxes so that you don't always find yourself forgetting to add things to the repo-based version. 

Anyway, here is the list that would help me:

- pink faceyspacey logo in the sidebar. it can still have the github URL, but it needs to have recognition for its creator. After all, RFR is the fastest growing in popularity on CodeSandBox :)
- pink faceyspacey logo in the search results more importantly
- get rid of that useless "New" sandbox :) ...woulda been nice to have been in that animated gif about the search results. The game's changing quick brother. Who's hot today won't be hot tomorrow. And who's on the comeup today (me and you) will be on top tomorrow. We're at key point in the history of React and programming in general. I got far more things coming. You help me, I help you. I'm always thinking of you and how to prop this up. There's a long long road for both of us. I can help you code a lot of features very quickly if it's what you want by the way. I basically understand the client app fully. But u gotta take the lead. I'm not sure if u even want it, which is totally fine too. However this is an opportunity that needs to go as fast as possible. After all, this is an improved webpackbin.com with the code either borrowed from Christian or learned from what he did *(knowhow is more important than code)*. If you don't want your tech commoditized you gotta iterate fast so nobody can beat you. It's not about closing the source on it (cuz u do need help until u can pay full time developers). However it is about achieving so much knowhow that nobody can easily compete. If you don't think with business in mind, someone that does will eat your lunch. You go to school full time after all. Some developers work on open source full time. You're an exceptional programmer and got very far very young. When did you start by the way? I didn't even start programming until I was 24. Either way, the world is watching, you should be coding this 24/7, no school, no trips. Just FYI. Developers have worked decades to have something of their own that works as well as this already producing money (was I your first customer by the way?). I'm talking really fucking good developers (far beyond you and me) that still have to work for employers to support their family. So make sure you appreciate what you have. If you don't you will fail to get most value out of having created it. What you also have is I assume the fact that you live at home?? Less bills, less work responsibility, etc. You won't always have that. Right now at this age you have something most young people don't recognize. I happen to think you do. I'm just sayin, maximize it. When your burn rate is $0, it means you can code all the time on whatever you want. And if you want my real advice, I would drop out of school. Plenty of developers with computer science degrees will lambaste me for saying that, but they likely just don't want to admit the significance of what you've done already and that you've done something they've failed to do: build an income-generating tool for developers with wide application already achieving wide use. That bothers them subconsciously. They want you in school to validate the path they took. I want u outa school to validate the path I took ;). The difference is I'm conscious of inner intentions. Only consciousness will shed light on the best decision for you. MY PREDICTION: you don't drop out, CodeSandBox won't become the iconic platform it would otherwise become. 

Anyway that's all beside the point. The point is the most important thing is your early adopter power users. And I'm clearly #1. If you wanna hone your craft, you take your #1 evangelists and prop them up as high as possible. I'm giving you the clue now because over the next 8 weeks there will be a codesandbox in each of the 3 articles I release per week for each of the new tools. And it will be promoted everywhere. My shit is only gonna grow. It's doin a slower burn than ur tools because of its nature (instant user-oriented app vs. tool which requires adoption from unrecognized developer). Every week that will change.

In short, what i would do is write technical plans for each of the Issues I've posted so I can do it for you. You can do that from ur phone in vietnam and come back to features. For example i want to build the forks modal, but based on last time i wanna make sure i do it exactly how u want it so i dont waste time. I know i was a little overzealous last time--i mainly just wanted to master different areas of ur app or i wouldnt have done it like that. But the point is you have an army if you lead it properly. Ur clearly particular with ur codebase and operating at a high level. So operate at the highest level and get others to do your work for you.

I need some of these issues fixed brother. I know how quickly they take you. Minutes for some of these. People kill to have a startup that is exploding in popularity in Silicon Valley. They spend every waking moment to fix/enhance every one of these things. And believe if u weren't in school or vietnam and at YC, that's what you would be doing. That your not is a mistake. Don't pick this path unless you're willing to kill. Life will test you, as it does. And you will likely have to make the decision u were asking on twitter weeks ago. When this gets even more popular it will become impossible to operate on the side. And by its rate of growth, it's a few weeks off. That's the primary reason I'm talking as frankly. I.e. in case you don't see it. Life is all about sacrifice. Make the sacrifice for this. It's worth it. It's working. *And this is rare.* Your story is that of your will.

**Comment by CompuIves** [comment]
Export to GitHub feature would be an awesome feature, but we first need to lay some groundwork before we can start with it.

The biggest one being that we need to set up a snapshot system, where users can make branches of their sandboxes so they can work open a pull request after doing several saves instead of one. I don't want to have a save == 1 commit, that will be very messy for users like me that press save after every change.

But, before I want to have a snapshot system we need to have a new kind of storage system, because snapshots will be expensive if we copy all files and folders over for every snapshot. There is a much more efficient and simple system invented for Git.

The idea is that every module will have an identifier based on its content (created using `sha256`) instead of the content. The actual content resides in a central table called `objects`, with ids mapped to content, this will make forking and snapshotting much cheaper, because there is no duplicate content and most of the time only 5% of the content changes between snapshots.

After snapshotting we need to work on a team system, so multiple users can work on a single sandbox and thus the git repo. We can just build a basic version at first because this is not really pressing, I'm thinking of adding contributors `has_many` to sandboxes.

After this we can work on committing changes to GitHub, either directly to master, or by opening a new branch. There are a lot more things that need to be built before (like UI/UX, server endpoints, git-extractor endpoints).

This feature will be a lot of work, but will be built regardless. I want to do it right in the beginning with this, because there is no turning back after we've released this feature. I'll start working on it somewhere in August. The editor has more pressing requests and issues at the moment which I find more important, both for approachability for contributions and extensibility.

Okay, so I did do some changes in the meantime. After I deploy somewhere this week you can add tags to GitHub sandboxes by adding tags to the `keywords` config in `package.json`. The main reason is that one of the big features of git sandboxes is that more people can work on it, and I don't want to limit tags to only one user. This will be especially useful when you can edit github on codesandbox, because then the `package.json` will just change after the edit. This week I'm also going to build a `Claim Sandbox` button where you can set yourself as author on GitHub sandboxes, this will put your name in the sidebar and it'll allow you to showcase the sandbox.

Of course I'm open to suggestions, and maybe you know an easier way to have intuitive GitHub integration, I'd love to know them. Also, let's open up a separate issue for the ShareView simplifications.

**Comment by SaraVieira** [comment]
closed by lack of comments in over a year
